Updates WooCommerce.com links to Woo.com

jeffikus opened the pull request:
Updates all instances of WooCommerce.com links in both the wp-admin and frontend. This includes the developer composer and package files, readme, and language files.

<!-- Reference any related issues or PRs here -->
Fixes request from @kevinbates as per p1704919648335339-slack-C02PXDRC67P 

### How to test the changes in this Pull Request:

1. Install and activate Storefront.
2. Navigate to /wp-admin/themes.php?page=storefront-welcome and check if outbound links point to woo.com instead of woocommerce.com
3. Navigate to /wp-admin/themes.php?theme=storefront and check if outbound links point to woo.com instead of woocommerce.com
4. Navigate to the frontend of your website and check that the "[Built with Storefront](https://woo.com/products/storefront/)." link navigate to the product page on woo.com correctly.

### Changelog

<!-- Add suggested changelog entry here. For example: -->

> Fix – Updates wp-admin and frontend credit links to point to Woo.com instead of WooCommerce.com

<!-- See [previous releases](https://github.com/woocommerce/storefront/releases) for more examples. -->

jeffikus committed:
Updates WooCommerce.com links to Woo.com

Updates all instances of WooCommerce.com links in both the wp-admin and frontend. This includes the developer composer and package files, readme, and language files.

danielwrobert commented on inc/storefront-template-functions.php:
```diff
@@ -139,9 +139,9 @@ function storefront_credit() {
 		if ( apply_filters( 'storefront_credit_link', true ) ) {
 			if ( storefront_is_woocommerce_activated() ) {
-				$links_output .= '<a href="https://woocommerce.com" target="_blank" title="' . esc_attr__( 'WooCommerce - The Best eCommerce Platform for WordPress', 'storefront' ) . '" rel="noreferrer nofollow">' . esc_html__( 'Built with Storefront &amp; WooCommerce', 'storefront' ) . '</a>.';
+				$links_output .= '<a href="https://woo.com" target="_blank" title="' . esc_attr__( 'WooCommerce - The Best eCommerce Platform for WordPress', 'storefront' ) . '" rel="noreferrer nofollow">' . esc_html__( 'Built with Storefront &amp; WooCommerce', 'storefront' ) . '</a>.';
```
Not a blocker but it was recommended to also update this text to read:

> Built with WooCommerce.

The reasoning was that it simplifies the text, and puts the emphasis on WooCommerce, with the URL pointing to our new landing page for WooCommerce.

See this thread for context: p1704919987116639/1704919648.335339-slack-C02PXDRC67P

jeffikus replied:
Ah I missed that. Updated now.

jeffikus replied:
Updated the language pot file as well.

danielwrobert replied:
I also notice the E2E tests were failing because of the change to the text in the footer credit.

 I took the liberty to update the test (in be6292d) to look for the updated output. Tests are passing now.

danielwrobert reviewed (approved):
I left one non-blocking comment about the footer note.

Aside from that, I spot-tested the new URLs to confirm they point to working pages.

Everything LGTM! 🚀

jeffikus committed:
Updates footer credit.

Changes footer credit from:
"Built with Storefront & WooCommerce."
To
"Built with WooCommerce."

jeffikus committed:
Updates language file.

Updates language file with new footer credit string.

danielwrobert committed:
Update E2E test for footer credit to reflect changes.
